internal/dag/dag.go

```diff
@@ -349,17 +375,24 @@ func (d *DAG) recompute() *dag {
 				}
 			}
 			vhost(host, 80).routes[r.path] = r
+			continue
 		}
-	}
 
-	_d := new(dag)
-	for _, vh := range _vhosts {
-		_d.roots = append(_d.roots, vh)
-	}
-	for _, svh := range _svhosts {
-		_d.roots = append(_d.roots, svh)
+		// otherwise, if the route is delegating to another ingressroute, find it and process it.
+		if route.Delegate.Name != "" {
+			namespace := route.Delegate.Namespace
+			if namespace == "" {
+				// we are delegating to another IngressRoute in the same namespace
+				namespace = ir.Namespace
+			}
+			dir, ok := d.ingressroutes[meta{name: route.Delegate.Name, namespace: namespace}]
```

Comment:
Probably for the next PR but if not found then should be an error status condition.

Reply:
Yep. I think we can handle this in the next PR.